test/e2e/flask/multichain-api/testHelpers.ts

```diff
@@ -105,3 +159,89 @@ export const getExpectedSessionScope = (scope: string, accounts: string[]) => ({
   notifications: KnownNotifications.eip155,
   accounts: accounts.map((acc) => `${scope}:${acc.toLowerCase()}`),
 });
+
+export const addAndAuthorizeAccount = async (driver: Driver): Promise<void> => {
+  const editButtons = await driver.findElements('[data-testid="edit"]');
+  await editButtons[0].click();
+  await driver.clickElement({ text: 'New account', tag: 'button' });
+  await driver.clickElement({ text: 'Add account', tag: 'button' });
+  await driver.delay(regularDelayMs);
+
+  /**
+   * this needs to be called again, as previous element is stale and will not be found in current frame
+   */
+  const freshEditButtons = await driver.findElements('[data-testid="edit"]');
+  await freshEditButtons[0].click();
+  await driver.delay(regularDelayMs);
+
+  const checkboxes = await driver.findElements('input[type="checkbox" i]');
+  await checkboxes[0].click(); // select all checkbox
+  await driver.delay(regularDelayMs);
+
+  await driver.clickElement({ text: 'Update', tag: 'button' });
+};
+
+/**
+ * Deselect all networks but Ethereum Mainnet through extension UI.
+ *
+ * @param driver - E2E test driver {@link Driver}, wrapping the Selenium WebDriver.
+ */
+export const uncheckNetworksExceptMainnet = async (
+  driver: Driver,
+): Promise<void> => {
+  const editButtons = await driver.findElements('[data-testid="edit"]');
+  await editButtons[1].click();
+  await driver.delay(regularDelayMs);
+
+  const networkListItems = await driver.findElements(
+    '.multichain-network-list-item',
+  );
+
+  for (const item of networkListItems) {
+    const network = await item.getText();
+    const checkbox = await item.findElement(By.css('input[type="checkbox"]'));
+    const isChecked = await checkbox.isSelected();
+
+    // we make sure to uncheck every other previously selected network other than Ethereum Mainnet
+    if (isChecked && !network.includes('Ethereum Mainnet')) {
+      await checkbox.click();
+      await driver.delay(regularDelayMs);
+    }
+  }
+  await driver.clickElement({ text: 'Update', tag: 'button' });
+};
+
+/**
+ * Will assert only the requested networks are selected, and all other are not.
+ *
+ * @param driver - E2E test driver {@link Driver}, wrapping the Selenium WebDriver.
+ * @param networkList - list of networks to assert selection for.
+ */
+export const assertOnlyRequestedNetworksAreSelected = async (
```

Comment:
perhaps this guy should be moved to a different file ?
Since it's purpose is not necessarily as a utility / helper function, but rather an actual assertion that goes on in the test (which I wanted to abstract away so the actual test didn't get too bloated and confusing, grouping this part in a separate function here seemed appropriate)

Reply:
Probably fine for now until we see a common pattern emerge? up to you!